api/app/controllers/api/v1/instructors_controller.rb

```diff
@@ -14,30 +14,44 @@ def index
             render_success(instructors_by_position)
         end
 
-        # POST /add_instructor
+        # POST /instructors AND /add_instructor
         def create
-            position = Position.find(params[:position_id])
-            instructor = position.instructors.new(instructor_params)
-            if instructor.save # passes Instructor model validation
-                render_success(instructors_by_position)
+            # if we passed in an id that exists, we want to update
+            if params[:id] && Instructor.exists?(params[:id])
+                update and return
+            end
+
+            if not params.include?(:position_id)
+                instructor = Instructor.new(instructor_params)
+                instructor_create(instructor)
             else
-                instructor.destroy!
-                render_error(instructor.errors, instructors_by_position)
+                position = Position.find(params[:position_id])
+                instructor = position.instructors.new(instructor_params)
```

Comment:
Does this syntax create a new instructor from the instructor_params? If my previous comment was correct, then I thought you could only do:
```
instructor = Instructor.new(instructor_params)
instructor_create(instructor, true)
position.instructors.append(instructor)
position.save!
```
To create a new instructor, then add a new instructor to the position.

Reply:
This syntax does create a new `instructor` from `instructor_params`. I'll check out if it works differently from the code you wrote, though.

Reply:
I'm just wondering if the position with a new instructor is really being saved. i know you save the instructor, but I don't know if you save the fact that this position now has a new instrucotr.

Reply:
I think the `save` is actually a misnomer. It's not really saving to the db, `save` actually just checks if the data matching the table constraints. `new` already inserted the data into the db.

Reply:
https://guides.rubyonrails.org/active_record_basics.html#create

According to the docs though new will not insert the data to the database? Did you try an `object.reload` to reload it from database and check that the changes have propagated?

Reply:
Okay. I'll check that.

Reply:
I tried the `reload`, but it seems like `save` is only really relevant when creating a table. If you update, say, `position.instructors` after creation, it does persist.